samples/core/Multitenancy/examplesingle/Program.cs

```diff
@@ -0,0 +1,123 @@
+﻿// See https://aka.ms/new-console-template for more information
+using ExampleSingle;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+
+Console.WriteLine("Single database and table example.");
+
+// grab some types as sample data
+#pragma warning disable CS8601 // Where clause guarantees non-nullable
+var types = AppDomain.CurrentDomain.GetAssemblies()
+    .SelectMany(a => a.ExportedTypes)
+    .Select(t => new { t.Namespace, t.Name })
+    .Where(t => !string.IsNullOrEmpty(t.Namespace))
+    .Select(t => new MultitenantType
+    {
+        Tenant = t.Namespace,
+        TypeName = t.Name
+    })
+    .ToList();
+#pragma warning restore CS8601 // Possible null reference assignment.
+
+// unique "tenants" = namespaces
+var tenants = types.Select(t => t.Tenant).Distinct()
+    .OrderBy(t => t).ToList<string>();
+
+var provider = new TenantProvider { Tenant = tenants[0] };
+
+// using this so the sample is relevant to ASP.NET Core apps
+var serviceProvider = new ServiceCollection()
+            .AddLogging()
+            .AddSingleton<TenantProvider>()
+            .AddDbContextFactory<SingleTenantContext>(
+                opt => opt.UseSqlite("Data Source=singleexample.sqlite")
+            )
+            .BuildServiceProvider();
+
+// prep the database. We're using "namespace" as tenant.
+var tenantProvider = serviceProvider.GetRequiredService<TenantProvider>();
+tenantProvider.Tenant = tenants[0];
+
+var factory = serviceProvider
+    .GetRequiredService<IDbContextFactory<SingleTenantContext>>();
+
+using (var ctx = factory.CreateDbContext())
+{
+    ctx.Database.EnsureDeleted();
+    ctx.Database.EnsureCreated();
+    ctx.MultitenantTypes.AddRange(types);
+    ctx.SaveChanges();
+}
+
+var done = false;
+
+while (!done)
```

Comment:
I'm not sure the loop with the int parsing helps - maybe keep it minimal with a quick code sample like in the other samples?